src/EFCore.PG/Infrastructure/Internal/NpgsqlOptionsExtension.cs

```diff
@@ -90,6 +94,94 @@ public override bool ApplyServices(IServiceCollection services)
+                = (RemoteCertificateValidationCallback?.GetHashCode() ?? 0).ToString(CultureInfo.InvariantCulture);
+
+            debugInfo["Npgsql.EntityFrameworkCore.PostgreSQL:" + nameof(NpgsqlDbContextOptionsBuilder.ProvideClientCertificatesCallback)]
+                = (ProvideClientCertificatesCallback?.GetHashCode() ?? 0).ToString(CultureInfo.InvariantCulture);
+
+            foreach (var rangeDefinition in _userRangeDefinitions)
+            {
+                debugInfo["Npgsql.EntityFrameworkCore.PostgreSQL:" + nameof(NpgsqlDbContextOptionsBuilder.MapRange) + ":" + rangeDefinition.SubtypeClrType.DisplayName()]
+                    = rangeDefinition.GetHashCode().ToString(CultureInfo.InvariantCulture);
+            }
+        }
+
+        [NotNull]
+        public override string LogFragment
+        {
+            get
+            {
+                if (_logFragment == null)
+                {
+                    var builder = new StringBuilder(base.LogFragment);
+
+                    if (AdminDatabase != null)
+                    {
+                        builder.Append("AdminDatabase=").Append(AdminDatabase).Append(' ');
+                    }
+
+                    if (PostgresVersion != null)
+                    {
+                        builder.Append("PostgresVersion=").Append(PostgresVersion).Append(' ');
+                    }
+
+                    if (ProvideClientCertificatesCallback != null)
+                    {
+                        builder.Append("ProvideClientCertificatesCallback ");
+                    }
+
+                    if (RemoteCertificateValidationCallback != null)
+                    {
+                        builder.Append("RemoteCertificateValidationCallback ");
+                    }
+
+                    if (ReverseNullOrdering)
+                    {
+                        builder.Append("ReverseNullOrdering ");
+                    }
+
+                    if (UserRangeDefinitions.Count > 0)
+                    {
+                        builder.Append("UserRangeDefinitions=[");
+                        foreach (var item in UserRangeDefinitions)
+                        {
+                            builder.Append(item.SubtypeClrType).Append("=>");
+
+                            if (item.SchemaName != null)
+                                builder.Append(item.SchemaName).Append(".");
+
+                            builder.Append(item.RangeName);
+
+                            if(item.SubtypeName != null)
+                                builder.Append("(").Append(item.SubtypeName).Append(")");
+
+                            builder.Append(";");
+                        }
+
+                        builder.Length = builder.Length -1;
+
+                        builder.Append("] ");
+                    }
+
+                    _logFragment = builder.ToString();
+                }
+                return _logFragment;
```

Comment:
You could invert the condition and decrease indentation if you wish.

Reply:
OK